New update function for incomplete products

5h3rr1ll opened the pull request:
The new update function only updates None fields

5h3rr1ll committed:
Merge branch 'dev'

5h3rr1ll committed:
Merge branch 'dev'

5h3rr1ll committed:
added another check, if brand already has a corp assigned

5h3rr1ll committed:
Merge branch 'hotfix🔥/AS_is_big_ten'

5h3rr1ll committed:
Added some print statements to the is_big_ten function

5h3rr1ll committed:
Merge branch 'hotfix🔥/AS_is_big_ten'

5h3rr1ll committed:
first commit for new check function

We need a extra function for the is_big_ten function to give a better quality feedback

5h3rr1ll committed:
added a else-case to the save_product-function for debugging reasons

5h3rr1ll committed:
created route for new update function

5h3rr1ll committed:
new update function for incomplete products

Darjusch commented on goodbuyDatabase/endpoints.py:
```diff
@@ -62,6 +70,13 @@ def create_feedback_string(request, product_object):
+    lst_string = string.split(",")
+
+    if BigTen.objects.filter(name=lst[0]).exists():
+        pass
```
wieso ist hier noch ein pass drinne?

Darjusch commented on goodbuyDatabase/endpoints.py:
```diff
@@ -179,9 +194,32 @@ def endpoint_save_product(request):
+        try:
+            product.name = json_product["name"]
+        except Exception as e:
+            print("name n.a.", str(e))
```
Exceptions müssten geloggt werden. 
Print statements an sich sollte nicht mehr im code stehen.

5h3rr1ll replied:
@Darjusch kein Plan wohin das geloggt werden soll. Ich wüsste nicht das wie sowas in place haben. Wenn doch, dann feel free to change that line of code und committe es. 😊

Darjusch commented on goodbuyDatabase/endpoints.py:
```diff
@@ -21,11 +21,19 @@ def is_in_own_database(request, code):
-    return BigTen.objects.filter(name__trigram_similar=brand.name).exists()
+    print(f"Trigram Similar Brand: {product.brand}")
+    print(f"New Brand: {brand}")
+    if brand.corporation is not None:
```
if brand.corporation ist das gleiche wie  if brand.corporation is not None

5h3rr1ll replied:
Bro, es gut um den Code zwischen line 202 bis 220.

Darjusch commented on goodbuyDatabase/endpoints.py:
```diff
@@ -179,9 +194,32 @@ def endpoint_save_product(request):
+            product.main_product_category = json_product["category"]
+        except Exception as e:
+            print("category n.a.", str(e))
+        product.save()
```
Es muss gecheckt werden ob das Produkt jetzt complete ist und wenn ja auf complete gesetzt werden.
Source -> Muss auf USER gesetzt werden.

5h3rr1ll replied:
Joa, das könnte man tun. Da das Produkt so oder so auf uncheck gesetzt wird, da derzeit eh manuell gecheckt werden muss, würde ich sagen, ist das jetzt erstmal die Aufgab der Person, die das Produkt checkt.

Darjusch commented on goodbuyDatabase/endpoints.py:
```diff
@@ -21,11 +21,19 @@ def is_in_own_database(request, code):
+        print(f"Brand Corp: {brand.corporation}")
+        print(f"Exists? {BigTen.objects.filter(name__trigram_similar=brand.corporation).exists()}")
+        return BigTen.objects.filter(name__trigram_similar=brand.corporation).exists()
+    return False
```
wenn wir keine brand.corporation haben ist es automatisch False --> sollte da nicht we dont know stehen?

5h3rr1ll replied:
Bro, es gut um den Code zwischen line 202 bis 220.

Darjusch reviewed (commented):
Please take a short look at my comments

5h3rr1ll committed:
Update endpoints.py

Vorschlag von Darjusch umgesetzt bzgl. line 32 "if brand.corporation is not None:"

5h3rr1ll committed:
Update endpoints.py

Umsetzung von @Darjusch Hinweis auf Zeile 36 "return False". Korrekt, wenn die if-Abfrage False returned sollte unsere Antwort "We don't know" lauten.

5h3rr1ll committed:
Update endpoints.py

Habe Funktion entfernt, die nichts mit diesem Pull-Request zu tun hat.
